cli/src/pcluster/resources/compute_node/user_data.sh

```diff
@@ -102,88 +102,110 @@ write_files:
+        echo ${!cookbook_version} | tee /opt/parallelcluster/.bootstrapped
+      fi
 
 --==BOUNDARY==
 Content-Type: text/x-shellscript; charset="us-ascii"
 MIME-Version: 1.0
 
 #!/bin/bash -x
 
+chmod +x /tmp/bootstrap.sh
+
 function error_exit
 {
-  instance_id=$(curl --retry 3 --retry-delay 0 --silent --fail http://169.254.169.254/latest/meta-data/instance-id)
-  log_dir=/home/logs/compute
-  mkdir -p ${!log_dir}
-  echo "Reporting instance as unhealthy and dumping logs to ${!log_dir}/${!instance_id}.tar.gz"
-  tar -czf ${!log_dir}/${!instance_id}.tar.gz /var/log
-  # TODO: add possibility to disable this behavior
-  # wait logs flush before signaling the failure
-  sleep 10
+  echo "Timed-out when bootstrapping instance"
+  sleep 10  # Allow logs to propagate
   shutdown -h now
   exit 1
 }
-function vendor_cookbook
-{
-  mkdir /tmp/cookbooks
-  cd /tmp/cookbooks
-  tar -xzf /etc/chef/aws-parallelcluster-cookbook.tgz
-  HOME_BAK="${!HOME}"
-  export HOME="/tmp"
-  for d in `ls /tmp/cookbooks`; do
-    cd /tmp/cookbooks/$d
-    LANG=en_US.UTF-8 /opt/cinc/embedded/bin/berks vendor /etc/chef/cookbooks --delete || error_exit 'Vendoring cookbook failed.'
-  done;
-  export HOME="${!HOME_BAK}"
-}
-[ -f /etc/profile.d/proxy.sh ] && . /etc/profile.d/proxy.sh
-custom_cookbook=${CustomChefCookbook}
-export _region=${AWS::Region}
-s3_url=${AWS::URLSuffix}
-if [ "${!custom_cookbook}" != "NONE" ]; then
-  if [[ "${!custom_cookbook}" =~ ^s3://([^/]*)(.*) ]]; then
-    bucket_region=$(aws s3api get-bucket-location --bucket ${!BASH_REMATCH[1]} | jq -r '.LocationConstraint')
-    if [[ "${!bucket_region}" == null ]]; then
-      bucket_region="us-east-1"
-    fi
-    cookbook_url=$(aws s3 presign "${!custom_cookbook}" --region "${!bucket_region}")
-  else
-    cookbook_url=${!custom_cookbook}
-  fi
-fi
-export PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin:/opt/aws/bin
-export parallelcluster_version=aws-parallelcluster-${ParallelClusterVersion}
-export cookbook_version=${CookbookVersion}
-export chef_version=${ChefVersion}
-export berkshelf_version=${BerkshelfVersion}
-if [ -f /opt/parallelcluster/.bootstrapped ]; then
-  installed_version=$(cat /opt/parallelcluster/.bootstrapped)
-  if [ "${!cookbook_version}" != "${!installed_version}" ]; then
-    error_exit "This AMI was created with ${!installed_version}, but is trying to be used with ${!cookbook_version}. Please either use an AMI created with ${!cookbook_version} or change your ParallelCluster to ${!installed_version}"
-  fi
+
+if [ "${Timeout}" == "NONE" ]; then
+  /tmp/bootstrap.sh
 else
-  error_exit "This AMI was not baked by ParallelCluster. Please use pcluster createami command to create an AMI by providing your AMI as parent image."
-fi
-if [ "${!custom_cookbook}" != "NONE" ]; then
-  curl --retry 3 -v -L -o /etc/chef/aws-parallelcluster-cookbook.tgz ${!cookbook_url}
-  vendor_cookbook
+  timeout ${Timeout} /tmp/bootstrap.sh || error_exit
```

Comment:
probably you have already checked, but please confirm that the `timeout`command is available by default in all the supported OSs

Reply:
Yes, tested for vanilla ami for alinux2, centos7, ubuntu18 and ubuntu20